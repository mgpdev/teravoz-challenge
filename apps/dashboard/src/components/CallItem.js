import React from "react";
import PropTypes from "prop-types";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import Avatar from "@material-ui/core/Avatar";
import PhoneRounded from "@material-ui/icons/PhoneRounded";

const styles = theme => ({
  root: {
    width: "100%"
  }
});

function CallItem(props) {
  const { number, time } = props;
  return (
    <div>
      <ListItem>
        <Avatar>
          <PhoneRounded />
        </Avatar>
        <ListItemText primary={number} secondary={time} />
      </ListItem>
    </div>
  );
}

CallItem.propTypes = {
  number: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired
};

export default CallItem;
