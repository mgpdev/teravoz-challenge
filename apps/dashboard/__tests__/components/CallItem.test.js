import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

// Component to be tested
import CallItem from "./../../src/components/CallItem";

describe("<CallItem />", () => {
  describe("render()", () => {
    test("renders the component", () => {
      const wrapper = shallow(
        <CallItem number="(11) 9986-5645" time="11:50" />
      );
      const component = wrapper.html();

      expect(component).toMatchSnapshot();
    });
  });
});
