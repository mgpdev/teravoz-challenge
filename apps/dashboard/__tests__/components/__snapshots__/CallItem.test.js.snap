// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CallItem /> render() renders the component 1`] = `"<div><li class=\\"MuiListItem-root-1 MuiListItem-default-4 MuiListItem-gutters-8\\"><div class=\\"MuiAvatar-root-12 MuiAvatar-colorDefault-13\\"><svg class=\\"MuiSvgIcon-root-15\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path><g><path d=\\"M19.23 15.26l-2.54-.29c-.61-.07-1.21.14-1.64.57l-1.84 1.84c-2.83-1.44-5.15-3.75-6.59-6.59l1.85-1.85c.43-.43.64-1.03.57-1.64l-.29-2.52c-.12-1.01-.97-1.77-1.99-1.77H5.03c-1.13 0-2.07.94-2 2.07.53 8.54 7.36 15.36 15.89 15.89 1.13.07 2.07-.87 2.07-2v-1.73c.01-1.01-.75-1.86-1.76-1.98z\\"></path></g></svg></div><div class=\\"MuiListItemText-root-22\\"><span class=\\"MuiTypography-root-28 MuiTypography-subheading-35 MuiListItemText-primary-25\\">(11) 9986-5645</span><p class=\\"MuiTypography-root-28 MuiTypography-body1-37 MuiTypography-colorTextSecondary-51 MuiListItemText-secondary-26\\">11:50</p></div></li></div>"`;
